modules/_labs/select/react/lib/SelectBase.tsx

```diff
@@ -363,105 +326,140 @@ export default class SelectBase extends React.Component<SelectBaseProps> {
-    const selectedOption = !isEmpty ? options[getCorrectedIndexByValue(options, value)] : null;
-    const selectedOptionLabel = selectedOption ? selectedOption.label : '';
-    const selectedOptionValue = selectedOption ? selectedOption.value : '';
-
-    return (
-      <SelectWrapper grow={grow} disabled={disabled}>
-        <SelectButton
-          aria-expanded={!isMenuHidden ? 'true' : undefined}
-          aria-haspopup="listbox"
-          aria-controls={!isMenuHidden ? this.menuId : undefined}
-          disabled={disabled}
+  // If the focused option changed, scroll the newly focused option into view (if
+  // necessary) but do NOT center it
+  useLayoutEffect(() => {
+    const focusedOption = focusedOptionRef.current;
+
+    if (focusedOption) {
+      // TODO: Figure out if rAF is the best approach here. I initially added
+      // rAF to get the Select States Menu On story to render correctly in IE.
+      // Without rAF, the menu is scrolled slightly further down than it should
+      // be (only in IE, and only in the Menu On visual testing stories), which
+      // triggers a visual regression. We're inside useLayoutEffect here so I
+      // didn't expect to need rAF in order to make proper measurements, but it
+      // seems to be necessary in IE.
+      //
+      // This rAF call also has the additional benefit of fixing a jarring menu
+      // placement issue in IE (https://github.com/Workday/canvas-kit/issues/791),
+      // so I'm leaving it in for now.
+      const animateId = requestAnimationFrame(() => {
+        // We cannot use the native Element.scrollIntoView() here because it
+        // doesn't work properly with the portalled menu: when using the keyboard
+        // to advance focus through the options, using scrollIntoView to keep the
+        // newly focused option in view also scrolls the ENTIRE page. Instead, we
+        // call our own scrollIntoViewIfNeeded function.
+        scrollIntoViewIfNeeded(focusedOption, false);
+      });
+
+      return () => {
+        cancelAnimationFrame(animateId);
+      };
+    }
+
+    return undefined;
+  }, [focusedOptionIndex]);
+
+  // If the menu was just opened, scroll the focused option into view (if
+  // necessary) and center it
+  useLayoutEffect(() => {
+    const focusedOption = focusedOptionRef.current;
+
+    // We need to scroll if the menu is either opening or opened in case we decide to
+    // bypass the opening state and jump straight to opened (like in visual testing,
+    // for instance)
+    if (focusedOption && (menuVisibility === 'opening' || menuVisibility === 'opened')) {
+      // TODO: This rAF call is also necessary for the Menu On visual testing
+      // stories to render properly in IE (see above, both rAF calls need to be
+      // present). It has no bearing on the menu placement issue.
+      const animateId = requestAnimationFrame(() => {
+        scrollIntoViewIfNeeded(focusedOption, true);
+      });
```

Comment:
This `rAF` call is also necessary for visual testing in IE (both `rAF` calls are needed to avoid a visual regression in IE).